pkg/platform/kube/platform.go

```diff
@@ -556,9 +556,28 @@ func (p *Platform) DeleteFunction(ctx context.Context, deleteFunctionOptions *pl
 		return nil
 	}
 
-	// user must clean api gateway before deleting the function
-	if err := p.validateFunctionHasNoAPIGateways(ctx, deleteFunctionOptions); err != nil {
-		return errors.Wrap(err, "Failed to validate that the function has no API gateways")
+	if !deleteFunctionOptions.DeleteApiGateways {
+		// user must clean api gateway before deleting the function
+		if err := p.validateFunctionHasNoAPIGateways(ctx, deleteFunctionOptions); err != nil {
+			return errors.Wrap(err, "Failed to validate that the function has no API gateways")
+		}
+	} else {
+		apiGatewayNames, err := p.getApiGatewaysForFunction(ctx, deleteFunctionOptions.FunctionConfig.Meta.Namespace, deleteFunctionOptions.FunctionConfig.Meta.Name)
+		if err != nil {
+			return errors.Wrap(err, "Failed to get a function to API-gateways mapping")
+		}
+		deleteAPIGatewayOptions := &platform.DeleteAPIGatewayOptions{
+			AuthSession: deleteFunctionOptions.AuthSession,
+		}
+		for _, apiGatewayInstance := range apiGatewayNames {
```

Comment:
you don't actually use the names at all, so we not return a list from `getApiGatewaysForFunction` instead of a map?